src/WP_CLI/SearchReplacer.php

```diff
@@ -114,19 +116,45 @@ private function _run( $data, $serialised, $recursion_level = 0, $visited_data =
 					$search_regex .= $this->regex_delimiter;
 					$search_regex .= $this->regex_flags;
 
-					$result = preg_replace( $search_regex, $this->to, $data, $this->regex_limit );
-					if ( null === $result || PREG_NO_ERROR !== preg_last_error() ) {
+					$check = true; // Set to avoid bogus error on strpos
+
+					if ( $this->callback ) {
+						if ( $check = strpos( $data, $this->from ) !== false ) {
+							$result = \call_user_func( $this->callback, $data, $this->to, $search_regex );
```

Comment:
Would passing `$this->form` (or `$this`) makes more sense thant `$search_regex`
1. Because callback may need raw `$from`
2. Because callback does `replace` which may rely on `regex_limit / flags` individually